engine/src/test/java/org/hibernate/validator/test/internal/engine/groups/sequence/SequenceTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.test.internal.engine.groups.sequence;
+
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertThat;
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.violationOf;
+
+import java.util.Set;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.GroupSequence;
+import javax.validation.Validator;
+import javax.validation.constraints.NotNull;
+
+import org.hibernate.validator.testutil.TestForIssue;
+import org.hibernate.validator.testutils.ValidatorUtil;
+
+import org.testng.annotations.Test;
+
+/**
+ * @author Marko Bekhta
+ */
+@TestForIssue( jiraKey = "HV-1715")
+public class SequenceTest {
+
+	@Test
+	public void groupSequenceOfGroupSequences() {
+		Validator validator = ValidatorUtil.getValidator();
+
+		Set<ConstraintViolation<Foo>> violations = validator.validate( new Foo( null, "", null ) );
+
+		assertThat( violations ).containsOnlyViolations(
+				violationOf( NotNull.class ).withProperty( "str1" )
+		);
+	}
+
+	interface Group1 extends Group11, Group12, Group13, Group14, Group15, Group16, Group17, Group18, Group19 {
```

Comment:
As that loop in which `validationSuccessful` is set is based on a `Set` it actually doesn't give consistent results. To be able to reproduce it more often I've added a lot of groups, which increase the probability that `true` will override the result in the end.